CONTRIBUTING.md

```diff
@@ -0,0 +1,98 @@
+# Contributing to Tangle-Accelerator
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a 
+   build.
+2. Create a new branch for fixing or developing the changes.
+3. Run test with `bazel test //tests/...`, after finishing the changes.
+4. Update the README.md with details of changes to the interface, this includes new environment 
+   variables, exposed ports, useful file locations and container parameters.
+5. Run `hooks/formatter` before committing the changes.
+6. Rebase to the latest `develop` branch.
+
+## Git Commit Message Guidelines
+Read this [blog article](https://chris.beams.io/posts/git-commit/) and [this article](https://www.conventionalcommits.org/en/v1.0.0-beta.2/) and follow the instructions in these articles.
+The subject line of git commit message should follow this pattern
+`<type>[optional scope]: <description>`
+The `type` includes the following 7 words depend on the content of the commit.
+
+* feat
+* fix
+* refactor
+* test
+* chore
```

Comment:
Did we really use `chore` before?

Reply:
We have only use 
`feat`, `fix`, `refactor` and maybe `doc`, `style` before

Reply:
Let's simplify the conventions.

Reply:
how about simplifying to 
* feat
* fix
* refactor
* test
* doc

Reply:
I defer to @YingHan-Chen .

Reply:
> how about simplifying to
> 
> * feat
> * fix
> * refactor
> * test
> * doc

I think those could cover most of the changes.

Reply:
then let me do it in this way